apps/passport-server/src/apis/devconnectPretixAPI.ts

```diff
@@ -0,0 +1,214 @@
+        });
+        if (!res.ok) {
+          throw new Error(
+            `[PRETIX] Error fetching ${url}: ${res.status} ${res.statusText}`
+          );
+        }
+        const page = await res.json();
+        items.push(...page.results);
+        url = page.next;
+      }
+
+      return items;
+    });
+  }
+
+  // Fetch all orders for a given event.
+  public async fetchOrders(
+    orgUrl: string,
+    token: string,
+    eventID: string
+  ): Promise<DevconnectPretixOrder[]> {
+    return traced(TRACE_SERVICE, "fetchOrders", async () => {
+      const orders: DevconnectPretixOrder[] = [];
+
+      // Fetch orders from paginated API
+      let url = `${orgUrl}/events/${eventID}/orders/`;
+      while (url != null) {
+        logger(`[DEVCONNECT PRETIX] Fetching ${url}`);
+        const res = await fetch(url, {
+          headers: { Authorization: `Token ${token}` },
+        });
+        if (!res.ok) {
+          throw new Error(
+            `[PRETIX] Error fetching ${url}: ${res.status} ${res.statusText}`
+          );
+        }
+        const page = await res.json();
+        orders.push(...page.results);
+        url = page.next;
+      }
+
+      return orders;
+    });
+  }
+}
+
+function pretixConfigDBToDevconnectPretixConfig(
+  pretixOrganizersDB: PretixOrganizersConfig[]
+): DevconnectPretixConfig {
+  return {
+    organizers: pretixOrganizersDB.map((organizerDB) => ({
+      id: organizerDB.id,
+      orgURL: organizerDB.organizer_url,
+      events: organizerDB.events.map((eventDB) => ({
+        id: eventDB.id,
+        eventID: eventDB.event_id,
+        activeItemIDs: eventDB.active_item_ids,
+      })),
+      token: organizerDB.token,
+    })),
+  };
+}
+
+export async function getDevconnectPretixConfig(
+  dbClient: Pool
+): Promise<DevconnectPretixConfig | null> {
+  try {
+    const pretixConfig = pretixConfigDBToDevconnectPretixConfig(
+      await fetchPretixOrganizers(dbClient)
+    );
+    logger("[DEVCONNECT PRETIX] read config: " + JSON.stringify(pretixConfig));
+    return pretixConfig;
+  } catch (e) {
+    logger(`[INIT] error while querying pretix organizer configuration: ${e}`);
+    return null;
+  }
+}
+
+export async function getDevconnectPretixAPI(
+  dbClient: Pool
+): Promise<IDevconnectPretixAPI | null> {
+  const config = await getDevconnectPretixConfig(dbClient);
+
+  if (config === null) {
+    return null;
+  }
+
+  const api = new DevconnectPretixAPI(config);
+  return api;
+}
+
+// A Pretix order. For our purposes, each order contains one ticket.
+export interface DevconnectPretixOrder {
+  code: string; // "Q0BHN"
+  status: string; // "p"
+  testmode: boolean;
+  secret: string;
+  email: string;
+  positions: DevconnectPretixPosition[]; // should have exactly one
+}
+
+export interface DevconnectPretixItem {
+  id: number; // corresponds to "item" field in DevconnectPretixPosition
+  name: {
+    en: string; // English name of item
```

Comment:
noting to self that language is important

Reply:
adding comment